executors/stable/executor.py

```diff
@@ -0,0 +1,154 @@
+import argparse, os, sys, glob
+from io import BytesIO
+import torch
+import numpy as np
+from PIL import Image
+from tqdm import tqdm, trange
+from itertools import islice
+from einops import rearrange
+import time
+from pytorch_lightning import seed_everything
+from torch import autocast
+from contextlib import nullcontext
+from typing import Dict
+from random import randint
+
+from ldm.util import instantiate_from_config
+from ldm.models.diffusion.ddim import DDIMSampler
+from ldm.models.diffusion.plms import PLMSSampler
+
+from jina import Executor, DocumentArray, Document, requests
+from omegaconf import OmegaConf
+
+
+class StableDiffusionConfig():
+    '''
+    Configuration for stable diffusion.
+    '''
+    ckpt = None
+    config = None
+    plms = False
+    seed = randint(0, 2 ** 64)
+    ddim_steps = 100
+    fixed_code = False
+    ddim_eta = 0.0
+    n_iter = 1
+    height = 512
+    width = 512
+    C = 4 # latent channels
+    f = 8 # downsampling factor
+    n_samples = 4
+    scale = 7.5 # unconditional guidance scale
+    precision = 'autocast'
+
+
+def chunk(it, size):
+    it = iter(it)
+    return iter(lambda: tuple(islice(it, size)), ())
+
+
+def load_model_from_config(config, ckpt, verbose=False):
+    print(f"Loading model from {ckpt}")
+    pl_sd = torch.load(ckpt, map_location="cpu")
+    if "global_step" in pl_sd:
+        print(f"Global Step: {pl_sd['global_step']}")
+    sd = pl_sd["state_dict"]
+    model = instantiate_from_config(config.model)
+    m, u = model.load_state_dict(sd, strict=False)
+    if len(m) > 0 and verbose:
+        print("missing keys:")
+        print(m)
+    if len(u) > 0 and verbose:
+        print("unexpected keys:")
+        print(u)
+
+    model.cuda()
+    model.eval()
+    return model
+
+
+class StableDiffusionGenerator(Executor):
+    opt: StableDiffusionConfig = None
+    config = None
+    model = None
+
+    def __init__(self, stable_path: str, **kwargs):
+        super().__init__(**kwargs)
+        self.opt = StableDiffusionConfig
+        self.opt.config = f'{stable_path}/configs/stable-diffusion/v1-inference.yaml'
+        self.opt.ckpt = f'{stable_path}/models/ldm/stable-diffusion-v1/model.ckpt'
+
+        self.config = OmegaConf.load(f"{self.opt.config}")
+        self.model = load_model_from_config(self.config, f"{self.opt.ckpt}")
+
+    @requests(on='/')
+    def txt2img(self, docs: DocumentArray, parameters: Dict, **kwargs):
+        num_images = max(1, min(9, int(parameters.get('num_images', 1))))
+        request_time = time.time()
+        opt = self.opt
+        opt.n_samples = num_images
+        seed_everything(opt.seed)
+
+        device = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
```

Comment:
this should be done at init time and not at every request

Reply:
Yeah, it was moved in latest (along with moving model init to ``__init__`) but I haven't pushed, working on img2img.